frontend/src/components/CreateNewSongbook.tsx

```diff
@@ -1,64 +1,100 @@
-import { Box, Button, Checkbox, Heading, Input, Text } from "@chakra-ui/react";
+import {
+  Box,
+  Button,
+  Flex,
+  FormLabel,
+  Input,
+  Switch,
+  Text,
+} from "@chakra-ui/react";
 import { useState } from "react";
 import { useNavigate } from "react-router-dom";
 import { createNewSongbook } from "../services/songs";
 
 export default function CreateNewSongbook() {
   const [maxSongs, setMaxSongs] = useState("");
-  const [title, setTitle] = useState("");
-  const [noodleMode, setNoodleMode] = useState(false);
+  const [title, setTitle] = useState<string>("");
+  const [actionVerb, setActionVerb] = useState<string>("DANCE");
+  const [isNoodleMode, setIsNoodleMode] = useState<boolean>(false);
```

Comment:
Just FYI - you don't need to specify the generic in these cases (`<string>`) because typescript can already infer them from the value you're passing to useState. You can double check this yourself by mousing over the return type for useState before and after adding the generic tag, it should not change. No need to change the code, just wanted to mention it.

Reply:
Excellent! It makes me curious how state is handled in those (presumably rare) cases where multiple types could be allowed. Or is that a state to be avoided at all costs?

Reply:
You can either let TS infer the types (in the case of something like `useState(myBool || "No")`, or you can explicitly give it a union type (`useState<string | boolean>("hi")`)